src/napari_clusters_plotter/_new_plotter_widget.py

```diff
@@ -745,3 +773,39 @@ def _reset(self):
         self._update_layer_colors(use_color_indices=False)
         self.control_widget.hue_box.setCurrentText("MANUAL_CLUSTER_ID")
         self.plot_needs_update.emit()
+
+    def _is_selectable(self, layer: napari.layers.Layer) -> bool:
+        """
+        Check if the layer is selectable.
+        """
+        if type(layer) in self.selectable_layer_types:
+            return True
+        return False
+
+    def _get_selected_objects(self, layer: napari.layers.Layer) -> List[int]:
+        """
+        Retrieve id of selected object on napari canvas"
+        """
+        if isinstance(layer, napari.layers.Points):
+            return list(layer.selected_data)
+        elif isinstance(layer, napari.layers.Labels):
+            return [layer.selected_label]
+        else:
+            raise TypeError(
+                f"Layer type {type(layer)} is not supported for selection."
+            )
```

Comment:
This doesn't necessarily have to be a member of the `PlotterWidget` - none of the functionality relies on `self`. Originally, some other functionality in here (e.g., `_set_layer_color`) weren't member functions but I think that was lost somewhere along the way. 

Still - moving it to `_utilities.py` could help to keep this widget a bit cleaner here :)